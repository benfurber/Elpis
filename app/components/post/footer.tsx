import React from "react";
import { StyleSheet, View } from "react-native";

import { Avatar } from "components";
import { Author } from "interfaces";
import { colours, layout } from "styles";

interface Props {
  avatarPath: Author["avatarPath"];
}

const Footer = (props: Props) => {
  return (
    <View style={styles.footer}>
      <View style={styles.followingContainer} />
      <View style={styles.avatarContainer}>
        <Avatar
          avatarPath={props.avatarPath}
          size={"large"}
          styles={styles.authorAvatar}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  authorAvatar: {
    borderColor: colours.navyBlueDark,
    borderWidth: 4,
    position: "absolute",
    top: -50,
  },
  avatarContainer: {
    alignItems: "center",
    flex: 1,
    flexDirection: "column",
  },
  followingContainer: {
    backgroundColor: colours.navyBlueDark,
    borderBottomLeftRadius: layout.borderRadius,
    borderBottomRightRadius: layout.borderRadius,
    flex: 1,
    padding: layout.spacingL,
  },
  footer: {
    backgroundColor: colours.whiteTransparent,
  },
});

export { Footer };
