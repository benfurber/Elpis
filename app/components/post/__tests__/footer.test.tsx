import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Footer } from "../footer";

describe("Footer", () => {
  it("renders correctly", () => {
    const component = shallow(
      <Footer
        avatarPath={require("assets/images/empower_two_women_logo.png")}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
