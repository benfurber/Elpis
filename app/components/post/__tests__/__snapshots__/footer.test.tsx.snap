// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post<Footer> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(45, 82, 113)",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 1,
        "padding": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Avatar
      avatarPath={
        Object {
          "testUri": "../../../app/assets/images/empower_two_women_logo.png",
        }
      }
      size="large"
      styles={
        Object {
          "borderColor": "rgb(45, 82, 113)",
          "borderWidth": 4,
          "position": "absolute",
          "top": -50,
        }
      }
    />
  </View>
</View>
`;
